### PR TITLE
Create install_with_git.sh script

### PR DESCRIPTION
**Pull request by anhlee66**
This script should do the following steps:
1. Check if git is installed (if not prompt the user to install git locally and have them create a GitHub account)
2. Create siglen-contrib and `git clone git@github.com:siglens/siglens.git` into this folder.
3. Go to this repository and run `go  mod tidy`
4. Check if current user is allowed to open. if yes do` go run cmd/siglens/main.go -config server.yaml` on port 80, else do `PORT=8090 go run cmd/siglens/main.go -config server.yaml`
5. Display a msg that SigLens is running on port [http://localhost:80](http://localhost/) or port http://localhost:8090/ depending on above 2 steps

### CODE REVIEW

File: install_with_git.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+cd siglens-contrib
+git clone git@github.com:siglens/siglens.git
+
+# Step 3: Change directory and download Go dependencies
+display_step 3 "Changing directory and downloading Go dependencies"
+cd siglens
+go mod tidy
+
+# Step 4: Check if the user is allowed to open port 80
+if [ "$(id -u)" = "0" ]; then
```
[the remark this answers is not in the document]
Comment (anhlee66) [reply install_with_git.sh]: I haven't found good solution for this, so I keep the other part and display msg when cannot access to localhost:80

File: install_with_git.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+mkdir -p siglens-contrib
+cd siglens-contrib
+git clone git@github.com:siglens/siglens.git
+
+# Step 3: Change directory and download Go dependencies
+display_step 3 "Changing directory and downloading Go dependencies"
+cd siglens
+go mod tidy
+
+# Step 4: Check if the user is allowed to open port 80
```
Comment (nkunal) [inline install_with_git.sh:27]: nit: remove this comment

File: install_with_git.sh [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+PORT=80 #Changing to PORT=8090 if able access to localhost:80
```
Comment (nkunal) [inline install_with_git.sh:29]: nit: comment not worded correctly

File: install_with_git.sh [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+display_step 4 "Running the Server on http://localhost:$PORT"
+if [ $PORT == 80 ]; then 
+	echo "If you are not able to access localhost:80 to changing the PORT=8090"
```
Comment (nkunal) [inline install_with_git.sh:33]: nit: comment not worded correctly

**Review by nkunal — COMMENTED** [review]
Sounds good, a couple of minor comments. Once done, can be merged

**Comment by anhlee66** [comment]
> Sounds good, a couple of minor comments. Once done, can be merged

I've removed this commit. This is my first issue so not done well.

**Review by nkunal — APPROVED** [review]

**Pull request merged**